From c3fa4699b9b761be1b406dcdd22344d56efd703b Mon Sep 17 00:00:00 2001
From: Nick Owens <mischief@coreos.com>
Date: Mon, 30 Nov 2015 18:48:05 -0800
Subject: [PATCH] partial revert of 3a1638dda19bbc73d0ae02b4c251ce08e564b4b9

in 3a1638dda19bbc73d0ae02b4c251ce08e564b4b9 from
git://anongit.mindrot.org/openssh.git, DSA (ssh-dss) key support was
removed. re-enable DSA key suport by adding back
ssh-dss-cert-v01@openssh.com and ssh-dss back to KEX_DEFAULT_PK_ALG.

this patch was generated with the following command on the above repo:

git diff -R 3a1638dda19bbc73d0ae02b4c251ce08e564b4b9^..3a1638dda19bbc73d0ae02b4c251ce08e564b4b9 -- myproposal.h
---
 myproposal.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/myproposal.h b/myproposal.h
index 46e5b98..371f27c 100644
--- a/myproposal.h
+++ b/myproposal.h
@@ -1,4 +1,4 @@
-/* $OpenBSD: myproposal.h,v 1.47 2015/07/10 06:21:53 markus Exp $ */
+/* $OpenBSD: myproposal.h,v 1.46 2015/07/03 03:47:00 djm Exp $ */
 
 /*
  * Copyright (c) 2000 Markus Friedl.  All rights reserved.
@@ -99,9 +99,11 @@
 	HOSTKEY_ECDSA_CERT_METHODS \
 	"ssh-ed25519-cert-v01@openssh.com," \
 	"ssh-rsa-cert-v01@openssh.com," \
+	"ssh-dss-cert-v01@openssh.com," \
 	HOSTKEY_ECDSA_METHODS \
 	"ssh-ed25519," \
-	"ssh-rsa" \
+	"ssh-rsa," \
+	"ssh-dss"
 
 /* the actual algorithms */
 
-- 
2.4.10

